File: iris-ld/iris-did/src/main/java/city/smartb/iris/did/DIDSigner.java
package city.smartb.iris.did;

import city.smartb.iris.crypto.dsl.signer.Signer;
import city.smartb.iris.ldproof.LdJsonObjectBuilder;
import city.smartb.iris.ldproof.LdProof;
import city.smartb.iris.ldproof.LdProofBuilder;
import city.smartb.iris.ldproof.crypto.RsaSignature2018LdProofSigner;

import java.security.GeneralSecurityException;

public class DIDSigner {

    public DIDDocument sign(DIDDocumentBuilder vcBuild, LdProofBuilder proofBuilder, Signer signer) throws GeneralSecurityException {
        RsaSignature2018LdProofSigner ldSigner = new RsaSignature2018LdProofSigner(signer, proofBuilder);
        LdJsonObjectBuilder builder = LdJsonObjectBuilder.builder(vcBuild.asJson());
        LdProof proof = ldSigner.sign(builder);
        return vcBuild.asJson(proof);
    }

}

File: iris-did/iris-did-lib/src/main/kotlin/city/smartb/iris/did/signer/DidSigner.kt
//package city.smartb.iris.did.signer
//
//import city.smartb.iris.crypto.rsa.RSAKeyPairReader
//import city.smartb.iris.did.DIDDocument
//import city.smartb.iris.signer.domain.features.SignQuery
//import city.smartb.iris.signer.domain.features.SignQueryFunction
//import f2.dsl.fnc.invoke
//import org.springframework.stereotype.Service
//
//@Service
//class DidSigner(
//    private val signQueryFunction: SignQueryFunction
//) {
//
//    suspend fun sign(didDocument: DIDDocument): DIDDocument {
//        val pair = RSAKeyPairReader.loadKeyPair("server")
//
//        val signedDidDocument = signQueryFunction.invoke(SignQuery(
//            jsonLd = didDocument,
//            privateKey = pair.private.toString(),
//            method = "rsa",
//            pathToVerificationKey = "todo path"
//        )).verifiableJsonLd
//
//        return DIDDocument(signedDidDocument.asJson())
//    }
//
//}
